jung-algorithms/src/main/java/edu/uci/ics/jung/algorithms/shortestpath/MinimumSpanningTree.java

```diff
@@ -111,14 +111,21 @@
       if (connectedNode == null) {
         tree.addNode(node);
       } else {
-        tree.putEdgeValue(node, connectedNode, graph.edgeValue(node, connectedNode));
+        tree.putEdgeValue(
+            node,
+            connectedNode,
+            graph.edgeValue(node, connectedNode).orElseThrow(IllegalStateException::new));
```

Comment:
Minor nit: if we're going to be (in effect) converting this from a `NoSuchElementException` to an `IllegalStateException`, we should add a message that at least tells the user that this is a bug in the library, not in their code.  Otherwise we're just replacing one mysterious exception with another.  :)

Reply:
Done. :)